### PR TITLE
Milestone 2: User interface

### PR DESCRIPTION
**Pull request by elpotatoslayer**
- Ask for the player's name
- Check if player names are identical. If they are, prompt the player to register for a different name
- Show the board to the user
- Write game instructions

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,3 +1,95 @@
...
+def player_log(player_name)
+  puts "please enter #{player_name} names. "
+  player_name = gets.chomp
+
+  while player_name.empty?
+    puts 'Enter a valid name'
+    player_name = gets.chomp
+  end
+
+  player_nameplayer_name
```
Comment (rindrajosia) [inline bin/main.rb:16]: Please check this line.

**Review by rindrajosia — CHANGES_REQUESTED** [review]
**Status: Changes Required**
Hello team!
Great job on your project. It does show you have an understanding of the game flow. There are however a few requirements needed before the project can be approved.

- [ ] [REQUIRED] game tells which player turn it is.
- [ ] [REQUIRED] game asks the player to select from available moves.
- [ ] [REQUIRED] game collects user input (move).
- [ ] [REQUIRED] game informs player if the selected move is invalid.
- [ ] [REQUIRED] game displays board after the player move.
- [ ] [REQUIRED] game informs player if the selected move is a winning move.
- [ ] [REQUIRED] game informs player if the selected move is a draw move.
- [ ] [REQUIRED] game repeats all actions for the next player's move.

**Implementation details**
- [ ] A loop is needed in this game.
- [ ] Inside the loop, you should call actions repeated for each move.

Eg:

> game_on = true
> 
>   =>        initial instructions for player
> 
> while game_on
>    =>        loop for each move
> 
>   if winner # and/or draw (the exact condition in this milestone is not important)
>     game_on = false
>   end
> end

Keep up with the efforts towards this project, you are almost there.
Rindra Josia

**Review by NtwaliHeritier — CHANGES_REQUESTED** [review]
### **STATUS: CHANGES REQUIRED**
Hello team,
Great work on the project, however some changes are required.
Check the changes I am going to show below.

File: bin/main.rb [hunk 2]
```diff
@@ -1,3 +1,116 @@
 #!/usr/bin/env ruby
+# rubocop:disable Metrics/BlockNesting
+require_relative 'board'
```
Comment (NtwaliHeritier) [inline bin/main.rb:3]: You are requiring a file that doesn't exist, and it is causing errors in running the codes.

File: bin/main.rb [hunk 3]
```diff
@@ -1,3 +1,116 @@
...
+def player_log(player_name)
+  puts "please enter #{player_name} names. "
+  player_name = gets.chomp
+
+  while player_name.empty?
+    puts 'Enter a valid name'
+    player_name = gets.chomp
+  end
+
+  player_nameplayer_name
```
Comment (NtwaliHeritier) [inline bin/main.rb:17]: On this line there is a typo, check it.

File: bin/main.rb [hunk 4]
```diff
@@ -1,3 +1,116 @@
...
+  [1, 5, 9],
+  [3, 5, 7]
+]
+
+# After setting up the board the game will then begin
+
+winner = false
+
+# The board will be shown for the players
+board = Board.new
```
Comment (NtwaliHeritier) [inline bin/main.rb:67]: This class Board is not available.

**Review by javitocor — APPROVED** [review]
# STATUS APPROVED 🥇 

Hello team! 👋 
You have met all the requirements 👍  and you have implemented all TSE suggestions! 💯 
Congratulations, project approved! 🥳 
Remember to merge it, fill up the completion form 📑  and you can move into the next milestone 🚀 
Good luck with it! 😺

**Pull request merged**